Mage.Sets/src/mage/cards/a/AshesOfTheFallen.java

```diff
@@ -62,7 +64,14 @@ public boolean apply(Game game, Ability source) {
                 for (UUID cardId : controller.getGraveyard()) {
                     Card card = game.getCard(cardId);
                     if (card != null && card.isCreature(game) && !card.hasSubtype(subType, game)) {
-                        game.getState().getCreateMageObjectAttribute(card, game).getSubtype().add(subType);
+                        MageObject mageObject = game.getObject(card.getId());
+                        if (mageObject != null) {
+                            CardUtil.getObjectPartsAsObjects(mageObject).forEach(objectPart ->{
+                                if (objectPart.isCreature(game)) {
```

Comment:
So you check isCreature here on the object part, is it really correct to check it earlier on the card also?